src/esp/gfx_batch/Hbao.h

```diff
@@ -0,0 +1,180 @@
+#ifndef ESP_GFX_BATCH_HBAO_H_
+#define ESP_GFX_BATCH_HBAO_H_
+
+#include <Corrade/Containers/Pointer.h>
+#include <Magnum/GL/GL.h>
+#include <Magnum/Magnum.h>
+#include <Magnum/Math/Vector2.h>
+
+namespace esp {
+namespace gfx_batch {
+
+enum class HbaoFlag {
+  Blur = 1 << 0,
+
+  /**
+   * Special blur is dependent on blur being set, and is ignored if blur is not
+   * set.
+   */
+  UseAoSpecialBlur = 1 << 1,
+  /* These two affect only the cache-aware variant. Only one can be set at
+     most, not both (mutually exclusive) */
+  LayeredImageLoadStore = 1 << 2,
+  LayeredGeometryShader = 1 << 3
+};
+typedef Corrade::Containers::EnumSet<HbaoFlag> HbaoFlags;
+
+CORRADE_ENUMSET_OPERATORS(HbaoFlags)
+
+class HbaoConfiguration {
+ public:
+  Magnum::Vector2i size() const { return size_; }
+
+  HbaoConfiguration& setSize(const Magnum::Vector2i& size) {
+    size_ = size;
+    return *this;
+  }
+
+  HbaoFlags flags() const { return flags_; }
+
+  HbaoConfiguration& setFlags(HbaoFlags flags) {
+    flags_ = flags;
+    return *this;
+  }
+
+  // Blur should always be used, otherwise there's nasty grid-like artifacts, at
+  // least until we support MSAA
+
+  // HbaoConfiguration& setUseBlur(bool state) {
+  //   // special blur is dependent on blur being enabled, so both should be
+  //   // disabled if blur is.
+  //   flags_ = (state ? flags_ | HbaoFlag::Blur
+  //                   : flags_ & ~(HbaoFlag::UseAoSpecialBlur |
+  //                   HbaoFlag::Blur));
+  //   return *this;
+  // }
```

Comment:
Should this be uncommented?

Reply:
> Blur should always be used, otherwise there's nasty grid-like artifacts, at least until we support MSAA

@jturner65 that's, as far as I understood, how it should be (the blur is what smoothens out the grid-like noise), and MSAA won't fix it. Ability to disable the blur is there for debugging purposes.